core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2098,11 +2098,55 @@ private[zio] trait ZIOFunctions extends Serializable {
    *
    * For a sequential version of this method, see `foreach`.
    */
-  final def foreachPar[R, E, A, B](as: Iterable[A])(fn: A => ZIO[R, E, B]): ZIO[R, E, List[B]] =
-    as.foldRight[ZIO[R, E, List[B]]](effectTotal(Nil)) { (a, io) =>
-        fn(a).zipWithPar(io)((b, bs) => b :: bs)
-      }
-      .refailWithTrace
+  final def foreachPar[R, E, A, B](as: Iterable[A])(fn: A => ZIO[R, E, B]): ZIO[R, E, List[B]] = {
+    val n         = as.size
+    val resultArr = new Array[Any](n)
+    val resultList: ZIO[R, E, List[B]] = for {
+      parentId <- ZIO.fiberId
+      latch    <- CountdownLatch.make(n)
+      failed   <- Promise.make[Nothing, Int]
+      cause    <- Ref.make[Cause[E]](Cause.empty)
+      fibers <- as
+                 .foldLeft[(Int, URIO[R, List[(Int, Fiber[E, B])]])]((0, URIO.succeed(Nil))) { (r, a) =>
```

Comment:
```suggestion
                 .foldLeft[(Int, URIO[R, List[(Int, Fiber[E, B])]])]((0, URIO.succeed(Nil))) { ((i, acc), a) =>
```

Reply:
Slightly embarrassingly, given how long I've been writing Scala, I didn't realize that was valid syntax...